### PR TITLE
[DFFS-40] Querying the Transportation Database

### PR DESCRIPTION
**Pull request by Mayowa2020**
Attached herewith are the screenshots of the results obtained from my queries.

Screenshot (396)
Screenshot (397)
Screenshot (398)
Screenshot (399)

### CODE REVIEW

File: submissions/BukkyOyetimehin/database/transportation_database.sql [hunk 1]
```diff
@@ -38,5 +50,54 @@ CREATE TABLE flight_statuses  (
...
+WHERE age < 27 AND (embarkation_port = 'Q' OR  embarkation_port = 'C');
+
+SELECT travel_details.passenger_detail_id, travel_details.embarkation_port, survivals.passenger_detail_id, survivals.survived FROM travel_details 
+JOIN survivals ON travel_details.passenger_detail_id = survivals.passenger_detail_id
+WHERE embarkation_port = 'S' AND survived = '1';
+
+SELECT passenger_details.id, passenger_details.full_name, passenger_details.age, travel_details.trip_fare, travel_details.sibsp_aboard FROM passenger_details 
+JOIN travel_details ON passenger_details.id = travel_details.passenger_detail_id
+WHERE  sibsp_aboard = '1' AND (trip_fare > '100' AND age > '35')
+
```
Comment (Ireti203) [inline submissions/BukkyOyetimehin/database/transportation_database.sql:90]: Is this supposed to be here?

**Review by adaugochi — CHANGES_REQUESTED** [review]
From line 1 - 72 should be a task you should have done before this task. Kindly do that first before doing this task. 

If you have done the previous task and you want to make any changes, make it on the previous task and push it as well.

**Comment by Ireti203** [comment]
You have still not made the requested changes @Mayowa2020. Querying the datatabase is a separate task and shouldn't be in the same file where you created the database.

File: submissions/BukkyOyetimehin/database/transportation_database_queries.sql [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+CREATE TABLE survivals (
+	id INT NOT NULL AUTO_INCREMENT,
+	passenger_detail_id INT NOT NULL,
+	survived BOOLEAN,       
+	PRIMARY KEY (id),	
+	FOREIGN KEY (passenger_detail_id) REFERENCES passenger_details (id)
+    );	    
+SHOW COLUMNS FROM survivals;
+
```
Comment (adaugochi) [inline submissions/BukkyOyetimehin/database/transportation_database_queries.sql:71]: You can now remove what you did here from 1 - 70.

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/BukkyOyetimehin/database/transportation_database_queries.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+-- 4. The total number of passengers that embarked at Southampton survived is 218.
+SELECT travel_details.passenger_detail_id, travel_details.embarkation_port, survivals.passenger_detail_id, survivals.survived FROM travel_details 
+JOIN survivals ON travel_details.passenger_detail_id = survivals.passenger_detail_id
+WHERE embarkation_port = 'S' AND survived = '1';
+
+-- 5. To get the id, name and the total number of passengers who paid a fare greater than $100 and above the age of 	35 had siblings or spouses on board:
+SELECT passenger_details.id, passenger_details.full_name, passenger_details.age, travel_details.trip_fare, travel_details.sibsp_aboard FROM passenger_details 
+JOIN travel_details ON passenger_details.id = travel_details.passenger_detail_id
+WHERE  sibsp_aboard = '1' AND (trip_fare > '100' AND age > '35');
+
```
Comment (adaugochi) [inline submissions/BukkyOyetimehin/database/transportation_database_queries.sql:26]: Remove spaces below

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by adaugochi — APPROVED** [review]

File: submissions/BukkyOyetimehin/database/transportation_database_queries.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+SELECT passenger_details.full_name, passenger_details.sex FROM passenger_details 
+LEFT JOIN travel_details ON passenger_details.id = travel_details.passenger_detail_id
+WHERE age < 27 AND (embarkation_port = 'Q' OR  embarkation_port = 'C');
+
+-- 4. The total number of passengers that embarked at Southampton survived is 218.
+SELECT travel_details.passenger_detail_id, travel_details.embarkation_port, survivals.passenger_detail_id, survivals.survived FROM travel_details 
+JOIN survivals ON travel_details.passenger_detail_id = survivals.passenger_detail_id
+WHERE embarkation_port = 'S' AND survived = '1';
+
+-- 5. To get the id, name and the total number of passengers who paid a fare greater than $100 and above the age of 	35 had siblings or spouses on board:
```
Comment (adaugochi) [inline submissions/BukkyOyetimehin/database/transportation_database_queries.sql:22]: I  didn't see the result of the `total number of passenger` for this question

**Review by adaugochi — DISMISSED** [review]

File: submissions/BukkyOyetimehin/database/transportation_database_queries.sql [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+-- 4. The total number of passengers that embarked at Southampton survived is 218.
+SELECT travel_details.passenger_detail_id, travel_details.embarkation_port, survivals.passenger_detail_id, survivals.survived FROM travel_details 
+JOIN survivals ON travel_details.passenger_detail_id = survivals.passenger_detail_id
+WHERE embarkation_port = 'S' AND survived = '1';
+
+-- 5. To get the id, name of passengers who paid a fare greater than $100 and above the age of 35 had and siblings or spouses on board:
+SELECT passenger_details.id, passenger_details.full_name, passenger_details.age, travel_details.trip_fare, travel_details.sibsp_aboard FROM passenger_details 
+JOIN travel_details ON passenger_details.id = travel_details.passenger_detail_id
+WHERE  sibsp_aboard = '1' AND (trip_fare > '100' AND age > '35');
+-- The total number of passengers who paid a fare greater than $100 and above the age of 35 and had siblings or spouses on board is 8.
```
Comment (adaugochi) [inline submissions/BukkyOyetimehin/database/transportation_database_queries.sql:26]: This answer is wrong. Are you sure this condition `sibsp_aboard = '1' AND (trip_fare > '100' AND age > '35')` is the right thing. Kindly check the question again

**Review by adaugochi — CHANGES_REQUESTED** [review]

File: submissions/BukkyOyetimehin/database/transportation_database_queries.sql (hunk 5 above)
Comment (Mayowa2020) [reply submissions/BukkyOyetimehin/database/transportation_database_queries.sql:26]: @adaugochi I have gone over the question again and again.  Not sure how else to write this query. Kindly assist with this, thank you.

Comment (adaugochi) [reply submissions/BukkyOyetimehin/database/transportation_database_queries.sql:26]: This section of the question says `had siblings or spouses on board?` . But, this what you are doing `sibsp_aboard = '1'`

**Review by adaugochi — CHANGES_REQUESTED** [review]

**Review by adaugochi — APPROVED** [review]

**Pull request merged**